src/main/java/com/es/phoneshop/model/recent_products/RecentProductsService.java

```diff
@@ -0,0 +1,12 @@
+package com.es.phoneshop.model.recent_products;
+
+import com.es.phoneshop.model.product.Product;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.Queue;
+
+public interface RecentProductsService {
+    Queue<Product> getRecentProducts(HttpServletRequest request);
```

Comment:
It's not really good to work with HttpServletRequest on service layer. If you want to simplify controller use utils and helpers instead of direct service invocation.